### PR TITLE
Add file status for Manuscript and Supporting uploads

### LINKED ISSUE #1374: Use File status on supporting file uploads
When uploading a supporting file(s) change the logic so that the status is set to `UPLOADED` only after it has been uploaded into S3 (+tests)

Also probably worth changing the manuscript upload too.
Part of #1362

### PR DESCRIPTION
**Pull request by diversemix**
Provides a file status for any uploaded file.
Closes #1374 
Tested locally and added/updated unit tests

### CODE REVIEW

File: server/xpub-server/entities/manuscript/resolvers/uploadManuscript.js [hunk 1]
```diff
@@ -117,7 +130,8 @@ async function uploadManuscript(_, { file, id, fileSize }, { user }) {
 
   // After the length file operations above - now update the manuscript...
   const manuscript = await Manuscript.find(id, userUuid)
-
+  if (manuscript.files[0].status !== 'STORED')
```
Comment (diversemix) [inline server/xpub-server/entities/manuscript/resolvers/uploadManuscript.js]: Need to remove

**Comment by diversemix** [comment]
Still to add the audit log for `CANCELLED`

**Review by hem-brahmbhatt — APPROVED** [review]

**Comment by diversemix** [comment]
@damnhipster ... The last commit removed the static function because this had issues in some of the supporting file test scenarios....  I found the workround was just to re-query for File and have added them in accordingly. You may wish to give it another once-over!

**Review by hem-brahmbhatt — APPROVED** [review]
Looks good, although I'd prefer to use a static `File.updateStatus` function to encapsulate this saving/finding logic we're having to introduce for the model

**Review by Cooryd — APPROVED** [review]

**Pull request merged**